### ISSUE TITLE
Optional parameter not patch url

### ISSUE THREAD

**Issue by riso348** [stein189/SimpleRouting#25]
Hi,
in previous version, was able to write this route:
`$router->add('/ajax/{a:action}{?:params}', 'GET', function ($action) ..`
and Router match 
```
/ajax/getModalContent/?ajax=true&action=getModalContent ...
/ajax/getModalContent?ajax=true&action=getModalContent ...
//ajax/getModalContent/?ajax=true&action=getModalContent ...
```

There was no problem in slashes, in v2 i have 
**Page not found**

Is it bug or its not allowed in new version?
Do i have to change urls in router to '/ajax/{a:action}**/**{?:params}' so every arguments will be separated by / ?

**Comment by stein189** [comment]
Thank you for your question,

You have to separate the optional parameter by a slash.
Everything after the ? (question mark) shouldnt be caught by the router, it should be handled by the $_GET global variable.

You can get the GET parameters from the $_GET global.

```
$router->add('/ajax/{a:action}', 'GET', function() {
    die(var_dump($_GET));
});
```

This should also match the following routes
```
/ajax/getModalContent/?ajax=true&action=getModalContent ...
/ajax/getModalContent?ajax=true&action=getModalContent ...
```

**Comment by riso348** [comment]
yes i change it in my routes but now i see something different:

```
$router->get('/ajax/{a:action}{?:params}', function ($action, $params = null) 

$router->get('/{*:page}', '\\Web\\WebManager::initializePage');
```
i call url /ajax/getModalContent?ajax=true&act
and the router accept second route with wildcard *:page instead of first route.

if i change first route into this:

`$router->get('/ajax/{a:action}', function ($action) `

router catch this route. Why, if the second attribute was optional, shouldnt he catch the first in both way, with optional parameter and without ?

**Comment by stein189** [comment]
@riso348 

Yes it should, it seems that the optional parameter does not work well in this particular case.

I will fix this problem in the next version.

**Comment by stein189** [comment]
This issue is solved in v2.0.3
